echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/impl/PipelineTriggerConverter.groovy

```diff
@@ -51,26 +52,27 @@ class PipelineTriggerConverter {
     return existingPipeline.withTrigger(trigger)
   }
 
-  static ActionInstance toScheduledAction(Pipeline pipeline, Trigger trigger) {
+  static ActionInstance toScheduledAction(Pipeline pipeline, Trigger trigger, String timeZoneId) {
     ActionInstance.ActionInstanceBuilder actionInstanceBuilder = ActionInstance.newActionInstance()
       .withId(trigger.id)
       .withName("Pipeline Trigger")
       .withGroup(pipeline.id)
       .withAction(PipelineTriggerAction)
-      .withParameters(toParameters(pipeline, trigger))
+      .withParameters(toParameters(pipeline, trigger, timeZoneId))
 
     if (Trigger.Type.CRON.toString().equalsIgnoreCase(trigger.type)) {
-      actionInstanceBuilder.withTrigger(new CronTrigger(trigger.cronExpression))
+      actionInstanceBuilder.withTrigger(new CronTrigger(trigger.cronExpression, timeZoneId, new Date()))
     }
 
     actionInstanceBuilder.build()
   }
 
-  static boolean isInSync(ActionInstance actionInstance, Trigger trigger) {
+  static boolean isInSync(ActionInstance actionInstance, Trigger trigger, String timeZoneId) {
     if (trigger.type == Trigger.Type.CRON.toString()) {
       return (
         actionInstance.trigger instanceof CronTrigger &&
-          trigger.cronExpression == ((CronTrigger) actionInstance.trigger).cronExpression
+          trigger.cronExpression == ((CronTrigger) actionInstance.trigger).cronExpression &&
+          timeZoneId == actionInstance?.context?.parameters?.triggerTimeZoneId
```

Comment:
I guess in the case of existing triggers where timeZoneId doesn't explicitly exist, they will just get updated here? (isInSync will be false the first time it is checked, which will re-persist the trigger including the configured timezone)

Also applies in the case of a persisted trigger with a timezone, and changing the system configured timezone?